internal/metrics/disk_stats_helper.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+type DiskStatsHelper interface {
+	GetDiskUsage(directory string) (uint64, uint64, error)
+}
+
+type OSDiskStatsHelper struct {
+}
+
+//go:generate mockgen -source=disk_stats_helper.go -package=metrics -destination=mock_disk_stats_helper.go
+func NewOSDiskStatsHelper() *OSDiskStatsHelper {
+	return &OSDiskStatsHelper{}
+}
+
+func (c *OSDiskStatsHelper) getUsedBytesInDirectory(directory string) (uint64, error) {
+	var totalBytes uint64
+	err := filepath.Walk(directory, func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !fileInfo.IsDir() {
+			totalBytes += uint64(fileInfo.Size())
+		}
+		return nil
+	})
+	return totalBytes, err
+}
+
+func (c *OSDiskStatsHelper) GetDiskUsage(directory string) (uint64, uint64, error) {
```

Comment:
Maybe named return values or a godoc block describing the function would help understand what are those unit64 that we are returning without reading the whole function.